### DISCUSSION TITLE
v2ray的inbound配置成https，curl可以请求，chrome不行

### DISCUSSION THREAD

**Discussion by cod1ingcoding** [v2fly/v2ray-core#2420] [Q&A]
v2ray的inbound配置成https，为什么curl可以，chrome的switchyomega报 ERR_HTTP2_PROTOCOL_ERROR，我的配置是:
```
{
  "log": {
    "loglevel": "warning"
  },
  "inbounds": [
    {
      "port": 8833, 
      "protocol": "http",
      "streamSettings": {
        "network": "tcp",
        "security": "tls",
        "tlsSettings": {
          "certificates": [
            {
              "certificateFile": "/etc/ssl/xxx.cn/fullchain.pem",
              "keyFile": "/etc/ssl/xxx.cn/privkey.pem"
            }
          ]
        }
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {}
    }
  ]
}
```

**Comment by mydogshitgold** [comment]
It seems that HTTP/2 transport (not the same as proxy) is not supported by V2Ray's HTTP proxy server.

1. You can force Chrome/Chromium based browsers to disable HTTP/2 with the `--disable-http2` flag to migrate this problem. Please note that disabling HTTP/2 will have a severe performance impact on complex web services.

2. Another option is to use tools that support HTTP/2 transport, such as [gost](https://gost.run/en/).

Please note, however, that HTTPS proxies are generally not suitable for bypassing an oppressive firewall, since they can be easily detected and blocked. In addition, HTTPS proxies can significantly increase handshake time, as the lack of multiplexing requires at least 2 additional round-trips for each new connection.

You may want to consider using other protocols.

**Comment by rurirei** [comment]
yes http2 is unsupported with a inbound of v2ray. 
>> It seems that HTTP/2 transport (not the same as proxy) is not supported by V2Ray's HTTP proxy server.
https://github.com/v2fly/v2ray-core/blob/master/transport/internet/http/hub.go

**Reply by xzycn** [reply]
Chrome/Chromium will do a http2 negotiation when you use https proxy, you can check it out [here](https://chromium.googlesource.com/chromium/src/+/HEAD/net/docs/proxy.md#https-proxy-scheme). 

The http2 negotiation will use ALPN extension(with h2 and http/1.1)，and v2fly has a default apln setting with **["h2","http/1.1"]**, v2fly server always specify the protocol to h2，so we can change it to "http/1.1" through changing alpn to **["http/1.1"]**  like as below:
```yaml
......
"tlsSettings": {
      "alpn": ["http/1.1"],
      "certificates": [
......
